src/components/query/QueryDirective.js

```diff
@@ -22,10 +25,11 @@ goog.require('ga_storage_service');
     $scope.queriesPredef = [];
     $scope.filters = [];
     $scope.featuresByLayer = {};
-    // return the year of the timestamp
-    var getYear = function(time) {
-      return (time && time.substr(0, 4) != '9999') ?
-          time.substr(0, 4) : undefined;
+
+    var getTimeInstantParam = function(layer) {
+      if (layer.timeEnabled) {
+        return gaTime.get() || gaTime.getYearFromTimestamp(layer.time);
```

Comment:
Yes it could but I prefer to keep this function as a simple parsing utilities function. May be we could need this without getting the current time.